plugins/catalog-backend/src/processors/UrlReaderProcessor.ts

```diff
@@ -130,9 +129,11 @@ export class UrlReaderProcessor implements CatalogProcessor {
   ): Promise<{ response: { data: Buffer; url: string }[]; etag?: string }> {
     // Does it contain globs? I.e. does it contain asterisks or question marks
     // (no curly braces for now)
+    // Azure DevOps uses the query string to contain the filepath so we need
+    // to evaluate the search part as well
+    const { pathname, search } = new URL(location);
 
-    const { filepath } = parseGitUrl(location);
-    if (filepath?.match(/[*?]/)) {
+    if (pathname?.match(/[*?]/) || search?.match(/[*?]/)) {
```

Comment:
We've spoken about moving this check to each of the URL reader implementations instead.

https://github.com/backstage/backstage/issues/22504

Maybe this is fine for now though? cc @Rugvip

Reply:
I think we need the more strict implementation here tbh, this seems very fragile still

Reply:
Yeah we rolled back this change before: https://github.com/backstage/backstage/pull/22141

Reply:
I've created https://github.com/backstage/backstage/pull/27054 which moves the condition to each URL Readers.

Is this what you had in mind @benjdlambert ?

When this gets merged, I'll update this PR to adapt the condition for the GerritUrlReader.